### ISSUE TITLE
Rule criteria: comparison with zero is broken

### ISSUE THREAD

**Issue by emptywee** [StackStorm/st2#2600]
Hi, I was working on the Icinga2 integration pack, and ran into the following issue:

```
2016-03-24 15:33:32,423 INFO [-] Validating rule icinga2.relay_to_chat for event.state_change.
2016-03-24 15:33:32,543 INFO [-] Validation for rule icinga2.relay_to_chat failed on criteria -
  key: trigger.state_type
  pattern: 0.0
  type: gt
  payload: 1.0
2016-03-24 15:33:32,544 INFO [-] 0 rule(s) found to enforce for event.state_change.
```

So when the Icinga2 API returns check results it uses float for numbers. For some reason rule engine doesn't match it correctly. I can work around it by stripping the fraction part in the sensor, but I think the problem is generally with the type conversions made in the stackstorm internally. 

Here's the payload:

```
INFO log [-] Validating rule icinga2.relay_to_chat for event.state_change. (trigger_instance={'occurrence_time': '2016-03-24 15:30:42.995794+00:00', 'trigger': u'icinga2.event.state_change', 'id': '56f408
238e326f157bf447c2', 'payload': {'check_result': {u'check_source': u'dc1udtlhtst02.stack.qadev.corp', u'exit_status': 1.0, u'vars_before': {u'state_type': 0.0, u'attempt': 2.0, u'state': 1.0, u'reachable': True}, u'schedule_start': 14588
33443.956044, u'vars_after': {u'state_type': 1.0, u'attempt': 1.0, u'state': 1.0, u'reachable': True}, u'execution_start': 1458833443.953009, u'schedule_end': 1458833443.956044, u'performance_data': [u'/=2547MB;4735;5327;0;5919', u'/boot
=105MB;154;173;0;193', u'/home=3MB;1560;1755;0;1951', u'/opt=111MB;15268;17177;0;19086', u'/tmp=1565MB;1560;1755;0;1951', u'/var=407MB;3224;3627;0;4031', u'/var/log=1684MB;3122;3512;0;3903'], u'state': 1.0, u'command': [u'/usr/lib64/nagi
os/plugins/check_disk', u'-c', u'10%', u'-w', u'20%', u'-K', u'10%', u'-W', u'20%', u'-X', u'none', u'-X', u'tmpfs', u'-X', u'sysfs', u'-X', u'proc', u'-X', u'devtmpfs', u'-X', u'devfs', u'-X', u'mtmfs', u'-m'], u'execution_end': 1458833
443.955872, u'active': True, u'output': u'DISK WARNING - free space: / 3064 MB (54% inode=84%); /boot 78 MB (42% inode=99%); /home 1845 MB (99% inode=99%); /opt 18002 MB (99% inode=99%); /tmp 283 MB (15% inode=99%); /var 3419 MB (89% ino
de=98%); /var/log 2014 MB (54% inode=99%);', u'type': u'CheckResult'}, 'service': u'disk', 'host': u'dc1udtlhtst02.stack.qadev.corp', 'state_type': 1.0, 'state': 1.0, 'type': u'StateChange'}},rule={'description': u'Post to chat when nagi
os service state changes', 'tags': [], 'ref': u'icinga2.relay_to_chat', 'enabled': True, 'name': u'relay_to_chat', 'trigger': u'icinga2.event.state_change', 'criteria': {u'trigger.type': {u'pattern': u'StateChange', u'type': u'eq'}, u'tr
igger.state_type': {u'pattern': u'0', u'type': u'gt'}}, 'action': 'ActionExecutionSpecDB@66037008(ref="chatops.post_message", parameters="{u\'message\': u\'Icinga2 service {{trigger.service}} on host {{trigger.host}} has changed its stat
e to {{ trigger.state|int }}, state type {{ trigger.state_type|int }} {{trigger.check_result.output}}\', u\'channel\': u\'563b5f7f21f7a36d7bd5baaf\'}")', 'pack': u'icinga2', 'type': 'RuleTypeSpecDB@66035856(ref="standard", parameters="{}
")', 'id': '56f2b7218e326f1e03ee5a1b', 'uid': u'rule:icinga2:relay_to_chat'},trigger={'uid': u'trigger:icinga2:event.state_change:5f02f0889301fd7be1ac972c11bf3e7d', 'parameters': {}, 'ref_count': 0, 'name': u'event.state_change', 'pack':
 u'icinga2', 'type': u'icinga2.event.state_change', 'id': '56f2b53e8e326f19afc50fdc', 'description': None})
```

**Comment by emptywee** [comment]
Turns out it was not related to floating points. Quoting the pattern in the rule definition converts numbers into strings and thus comparison doesn't work. Removing the quotes seems to help. However, 0 (without quotes) in the pattern is translated into `None` type and breaks the comparison.

```
2016-03-25 16:54:15,602 INFO [-] Validation for rule icinga2.relay_to_chat failed on criteria -
  key: trigger.state_type
  pattern: None
  type: gt
  payload: 1
2016-03-25 16:54:15,603 INFO [-] 0 rule(s) found to enforce for event.state_change.
```
criteria:
```
  trigger.state_type:
    pattern: 0
    type: gt
```

Dirty workaround is placing `0.0001` in the pattern. Might need to get inspected and fixed though.

**Comment by Kami** [comment]
Good catch.

I need to dig in, but I believe it's probably related to something in our code (we don't do a strict check for false values or similar) and not the actual YAML handling.

**Comment by Kami** [comment]
Yeah, I just managed to track down the bug with rule test tool :)

I will work on and push a fix later today.

**Comment by Kami** [comment]
I merged fix into master - #2602.

Thanks again for the report and a good catch!
